[Core] Brep Surface - trimmed integration (Part 2 - Inner loop)

rickyaristio opened the pull request:
**📝 Description**
This is the second part of the PR series to fix the problem related to the trimmed integration in the IGA application.
This PR adds additional operations for the inner loop trimming.

Dependencies: #13072 

The test cases will be elaborated on in the next PR.

rickyaristio committed:
add the operation for inner trimming

juancamarotti commented on kratos/utilities/geometry_utilities/brep_trimming_utilities.cpp:
Some comments:
1) Why not "solution_outer" instead of "solution" for more clarity and consistence?
2) Can we propose a better name for the Clipper operation, rather than "d"?

juancamarotti commented on kratos/utilities/geometry_utilities/brep_trimming_utilities.h:
Just for the sake of consistency, I would propose separating all the composed names with _, for example: best_vertex instead of bestvertex, or min_weight instead of minweight

juancamarotti reviewed (commented):
Just some minor comments regarding format...

juancamarotti committed:
Making the naming more consistence

We change "solution" to "solution_outer" and add a proper naming for the inner and outer Clipper operations

juancamarotti committed:
Making the naming more consistent

juancamarotti reviewed (dismissed):
Ok from my side 👍

rickyaristio committed:
Merge remote-tracking branch 'origin/master' into core/iga_trimmed_integration_inner
